txscript/opcode.go

```diff
@@ -2214,13 +2212,12 @@ func opcodeCheckMultiSig(op *parsedOpcode, vm *Engine) error {
 			}
 
 			// Parse the signature.
-			var err error
-			parsedSig, err = ecc.ParseSignature(signature)
-
-			sigInfo.parsed = true
+			parsedSig, err = secp256k1.DeserializeSchnorrSignatureFromSlice(signature)
 			if err != nil {
-				continue
+				return err
```

Comment:
How sure are you this is a good idea?

To me it looks correct as well, but this is the sort of changes I'd like to do really carefully, and preferably not as part of bigger changes.

So unless you know for an absolute fact that this is the correct behavior, I'd rather revert for now and then check it carefully at more peaceful times.

Reply:
It's good that you pointed this out.
I should really go over the consensus code again to check for correctness instead of workness

Reply:
it is actually *wrong* :) this will break op_checkmultisig

Reply:
By "this" you mean "what have been there up until now" or "my change"?

Reply:
arghh not really. sorry for spamming let's get this straight:
if you have: P1,P2,P3 in a 1/3 multisig, and you want to sign with the third key, then you need to provide: 
0*32,0*32,SIG. 
meaning the first 2 signatures will be invalid. but here the error is just about the length which is checked the line above so can never fail. non the less, this is a bad idea because if the behavior of `DeserializeSchnorrSignatureFromSlice` will ever change to actually do some verification then this will break op_checkmultisig.